protocol/x/revshare/keeper/revshare.go

```diff
@@ -142,3 +146,123 @@ func (k Keeper) ValidateRevShareSafety(
 	return totalRevSharePpm < lib.OneMillion
 }
+
+func (k Keeper) GetAllRevShares(
+	ctx sdk.Context,
+	fill clobtypes.FillForProcess,
+) ([]types.RevShare, error) {
+	revShares := []types.RevShare{}
+	totalFeesShared := big.NewInt(0)
+	takerFees := fill.TakerFeeQuoteQuantums()
+	makerFees := fill.MakerFeeQuoteQuantums()
+	netFees := big.NewInt(0).Add(takerFees, makerFees)
+
+	affiliateRevShares, affiliateRevFeesShared, err := k.getAffiliateRevShares(ctx, fill)
+	if err != nil {
+		return nil, err
+	}
+
+	unconditionalRevShares, unconditionalRevFeesShared, err := k.getUnconditionalRevShares(ctx, netFees)
+	if err != nil {
+		return nil, err
+	}
+
+	marketMapperRevShares, marketMapperRevFeesShared, err := k.getMarketMapperRevShare(ctx, fill.FillSourceId(), netFees)
+	if err != nil {
+		return nil, err
+	}
+
+	revShares = append(revShares, affiliateRevShares...)
+	revShares = append(revShares, unconditionalRevShares...)
+	revShares = append(revShares, marketMapperRevShares...)
+
+	totalFeesShared.Add(totalFeesShared, affiliateRevFeesShared)
+	totalFeesShared.Add(totalFeesShared, unconditionalRevFeesShared)
+	totalFeesShared.Add(totalFeesShared, marketMapperRevFeesShared)
+
+	//check total fees shared is less than or equal to net fees
+	if totalFeesShared.Cmp(netFees) > 0 {
+		return nil, errorsmod.Wrap(types.ErrTotalFeesSharedExceedsNetFees, "total fees shared exceeds net fees")
+	}
+
+	return revShares, nil
+}
+
+func (k Keeper) getAffiliateRevShares(
+	ctx sdk.Context,
+	fill clobtypes.FillForProcess,
+) ([]types.RevShare, *big.Int, error) {
+	takerAddr := fill.TakerAddr()
+	takerFee := fill.TakerFeeQuoteQuantums()
+	if fill.MonthlyRollingTakerVolumeQuantums() >= types.Max30dRefereeVolumeQuantums {
+		return nil, big.NewInt(0), nil
+	}
+
+	takerAffiliateAddr, feeSharePpm, exists, err := k.affiliatesKeeper.GetTakerFeeShare(ctx, takerAddr)
+	if err != nil {
+		return nil, big.NewInt(0), err
+	}
+	if !exists {
+		return nil, big.NewInt(0), nil
+	}
+	feesShared := lib.BigMulPpm(takerFee, lib.BigU(feeSharePpm), false)
+	return []types.RevShare{
+		{
+			Recipient:         takerAffiliateAddr,
+			RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_TAKER_FEE,
+			RevShareType:      types.REV_SHARE_TYPE_AFFILIATE,
+			QuoteQuantums:     feesShared,
+		},
+	}, feesShared, nil
+}
+
+func (k Keeper) getUnconditionalRevShares(
+	ctx sdk.Context,
+	netFees *big.Int,
+) ([]types.RevShare, *big.Int, error) {
+	revShares := []types.RevShare{}
+	totalFeesShared := big.NewInt(0)
+	unconditionalRevShareConfig, err := k.GetUnconditionalRevShareConfigParams(ctx)
+	if err != nil {
+		return nil, big.NewInt(0), err
+	}
+	for _, revShare := range unconditionalRevShareConfig.Configs {
+		feeShared := lib.BigMulPpm(netFees, lib.BigU(revShare.SharePpm), false)
+		totalFeesShared.Add(totalFeesShared, feeShared)
+		revShare := types.RevShare{
+			Recipient:         revShare.Address,
+			RevShareFeeSource: types.REV_SHARE_FEE_SOURCE_NET_FEE,
```

Comment:
On second thought, do we need this in any downstream logic?

Reply:
Dont we need revshare Fee source in [trading rewards formula](https://www.notion.so/dydx/RevShare-Generalization-6d89820112c94dfeaefc5a5ace052cf4?pvs=4#893602cca8b945548006414c877d2ed8)?

Reply:
And revshare type to retrieve affiliate revshares later for sending indexer events

Reply:
> Dont we need revshare Fee source in [trading rewards formula](https://www.notion.so/dydx/RevShare-Generalization-6d89820112c94dfeaefc5a5ace052cf4?pvs=4#893602cca8b945548006414c877d2ed8)?

Ah yes we need so that we can separate reward logic for maker and taker fee